### PR TITLE
CORTX-31644 removed the java-1.8.0-openjdk

### PR DESCRIPTION
**Pull request by Abhirup-A-G**
Signed-off-by: Abhirup Ghosh <abhirup.ghosh@seagate.com>

# Problem Statement
- CORTX-31644 removed the java-1.8.0-openjdk

# Design
- Removing the java-1.8.0-openjdk from the below files. 
In docker/cortx-build/dockerfile

https://github.com/Seagate/cortx-re/blob/c124e0fc5fddbc619fb2a279cdc66dbd58b6fabe/docker/cortx-build/Dockerfile#L70

In scripts/custom-iso/custom-packages.txt

https://github.com/Seagate/cortx-re/blob/c124e0fc5fddbc619fb2a279cdc66dbd58b6fabe/scripts/custom-iso/custom-packages.txt#L34

In scripts/custom-prereq/kafka/kafka.spec

https://github.com/Seagate/cortx-re/blob/c124e0fc5fddbc619fb2a279cdc66dbd58b6fabe/scripts/custom-prereq/kafka/kafka.spec#L15

In scripts/deployment/roles/vm_deployment/templates/cortx_deploy_prereq.sh.j2

https://github.com/Seagate/cortx-re/blob/c124e0fc5fddbc619fb2a279cdc66dbd58b6fabe/scripts/deployment/roles/vm_deployment/templates/cortx_deploy_prereq.sh.j2#L34

In scripts/third-party-rpm/install-cortx-prereq.sh

https://github.com/Seagate/cortx-re/blob/c124e0fc5fddbc619fb2a279cdc66dbd58b6fabe/scripts/third-party-rpm/install-cortx-prereq.sh#L40
https://github.com/Seagate/cortx-re/blob/c124e0fc5fddbc619fb2a279cdc66dbd58b6fabe/scripts/third-party-rpm/install-cortx-prereq.sh#L42

In solution/ova/roles/vm-provisioner/files/provisioner.sh

https://github.com/Seagate/cortx-re/blob/c124e0fc5fddbc619fb2a279cdc66dbd58b6fabe/solutions/ova/roles/vm-provisioner/files/provisioner.sh#L18

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing : 
- https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/Cortx-Automation/job/docker-image-ci/567/
- https://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/setup-kubernetes-cluster/4687/
- https://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/deploy-cortx-k8-cluster/1498/
- https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/8465/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [x] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/custom-prereq/kafka/kafka.spec [hunk 1]
```diff
@@ -12,7 +12,6 @@ URL:            https://kafka.apache.org/
 Source0:	kafka_%{_kafka_version}-%{_kafka_release}.tgz
 Source1:	kafka.service
 Source2:	kafka-zookeeper.service
-Requires:	java-1.8.0-openjdk-headless
```
Comment (shailesh-vaidya) [inline scripts/custom-prereq/kafka/kafka.spec:15]: Have we tested Kafka package generation and installation with this. If not we can skip this change

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: scripts/custom-prereq/kafka/kafka.spec (hunk 1 above)
Comment (Abhirup-A-G) [reply scripts/custom-prereq/kafka/kafka.spec:15]: Not tested. Skipping it.

Comment (Abhirup-A-G) [reply scripts/custom-prereq/kafka/kafka.spec]: Done

File: docker/cortx-build/Dockerfile [hunk 2]
```diff
@@ -67,7 +67,6 @@ jq \
 wget \
 which \
 openssh-server \
-java-1.8.0-openjdk \
```
[the remark this answers is not in the document]
Comment (Abhirup-A-G) [reply docker/cortx-build/Dockerfile:70]: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/AbhirupSpace/job/community_build_job2_/43/
you might see the job failed but the purpose of the job is to check if the community build is successful or not. So the community build is successful but there is some issue with the deployment due to which services fail to start. !n deployment is not effected by this removal of the java-1.8.0-openjdk. we can refer the below job : https://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/setup-kubernetes-cluster/4687/
https://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/deploy-cortx-k8-cluster/1498/
https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/8465/

[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply docker/cortx-build/Dockerfile:70]: @Abhirup-A-G - Can you please confirm that deployment is not failed because of the removal of java-1.8.0-openjdk

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

**Pull request closed without merging**